server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetVacationResponseMethod.java

```diff
@@ -42,6 +64,73 @@
 
     @Override
     public Stream<JmapResponse> process(JmapRequest request, ClientId clientId, MailboxSession mailboxSession) {
-        throw new NotImplementedException();
+        Preconditions.checkNotNull(request);
+        Preconditions.checkNotNull(clientId);
+        Preconditions.checkNotNull(mailboxSession);
+        Preconditions.checkArgument(request instanceof SetVacationRequest);
+        SetVacationRequest setVacationRequest = (SetVacationRequest) request;
+
+        if (!validateRequestStructure(setVacationRequest)) {
+            return Stream.of(JmapResponse
+                .builder()
+                .clientId(clientId)
+                .error(ErrorResponse.builder()
+                    .type(INVALID_ARGUMENTS1)
+                    .description(INVALID_ARGUMENT_DESCRIPTION)
+                    .build())
+                .build());
+        }
+
+        return process(clientId,
+            AccountId.create(mailboxSession.getUser().getUserName()),
+            setVacationRequest.getUpdate().get(Vacation.ID));
+    }
+
+    private boolean validateRequestStructure(SetVacationRequest setVacationRequest) {
+        return setVacationRequest.getUpdate().entrySet().size() == 1 && setVacationRequest.getUpdate().containsKey(Vacation.ID);
+    }
+
+    private Stream<JmapResponse> process(ClientId clientId, AccountId accountId, VacationResponse vacationResponse) {
+        boolean isValid = isValid(vacationResponse);
+
+        if (isValid) {
+            vacationRepository.modifyVacation(accountId, convertToVacation(vacationResponse));
+        }
+
+        return Stream.of(JmapResponse.builder()
+            .clientId(clientId)
+            .responseName(RESPONSE_NAME)
+            .response(generateSetVacationResponse(isValid, vacationResponse))
+            .build());
+    }
```

Comment:
I think if you split the two cases valid/not valid this method would be simpler to read and you won't need to pass a boolean as parameter